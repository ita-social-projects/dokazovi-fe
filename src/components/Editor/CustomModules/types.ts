import Quill from 'quill';
import { DropEvent, FileRejection } from 'react-dropzone';

export interface IUrlInputModalProps {
  editor?: Quill;
  updateBackgroundImage?: (url: string) => void;
  forBackgroundImg?: boolean;
}

export interface IBackgroundImageContainerProps {
  dispatchImageUrl: (backgroundImageUrl: string) => void;
  fileSelectorHandler: <T extends File>(
    acceptedFiles: T[],
    fileRejections: FileRejection[],
    event: DropEvent,
  ) => void;
  title?: string;
  imgUrl?: string;
  notCarousel?: boolean;
  reminder?: boolean;
  forMobilePic: boolean;
}

export interface IFileInputProps {
  onDrop: <T extends File>(
    acceptedFiles: T[],
    fileRejections: FileRejection[],
    event: DropEvent,
  ) => void;
  forMobilePic: boolean;
}
