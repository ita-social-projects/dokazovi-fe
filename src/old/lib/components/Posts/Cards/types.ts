import { IPost } from '../../../types';

export interface IPostPreviewCardProps {
  post: IPost;
  shouldNotUseLink?: boolean;
}
