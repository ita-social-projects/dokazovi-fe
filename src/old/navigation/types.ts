export interface IRouteConfig {
  component: React.ComponentType;
  path: string | string[];
  key: string;
  exact?: boolean;
  private?: boolean;
  useRender?: boolean;
}
