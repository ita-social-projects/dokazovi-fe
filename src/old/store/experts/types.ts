import { LoadingStatusEnum } from '../../lib/types';
import { IMaterials } from '../materials/types';

export interface IExpertPayload {
  expertIds: number[];
  meta: IExpertMeta;
}

export interface IExpertMeta {
  totalPages: number;
  pageNumber: number;
  loading: LoadingStatusEnum;
  error: null | string;
  isLastPage: boolean;
  totalElements: number;
}

export interface IExpertsState {
  experts: IExpertPayload;
  materials: IMaterials;
}

export interface IFetchExpertsOptions {
  page: number;
  regions: number[];
  directions: number[];
  appendExperts: boolean;
}
