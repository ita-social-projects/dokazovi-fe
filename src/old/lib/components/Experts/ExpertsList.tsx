import { Grid } from '@material-ui/core';
import React, { useEffect, useRef, useState } from 'react';
import { LOAD_EXPERTS_LIMIT } from '../../constants/experts';
import { IExpert } from '../../types';
import ExpertPhotoDataCard from './ExpertPhotoDataCard';

export interface IExpertsListProps {
  experts: IExpert[];
}

export const ExpertsList: React.FC<IExpertsListProps> = ({ experts }) => {
  const expertIdxForScroll = experts.length - LOAD_EXPERTS_LIMIT;
  const expertForScrollRef = useRef<HTMLDivElement>(null);

  const [prevExpertsCount, setPrevExpertsLength] = useState(experts.length);

  useEffect(() => {
    if (!expertForScrollRef.current) return;
    if (
      experts.length > LOAD_EXPERTS_LIMIT &&
      experts.length !== prevExpertsCount
    ) {
      expertForScrollRef.current.scrollIntoView({
        behavior: 'smooth',
      });
    }
    setPrevExpertsLength(experts.length);
  }, [experts.length]);

  return (
    <Grid container spacing={3}>
      {experts.map((expert, idx) => (
        <Grid
          item
          md={4}
          lg={4}
          key={expert.id}
          ref={expertIdxForScroll === idx ? expertForScrollRef : null}
        >
          <ExpertPhotoDataCard expert={expert} key={expert.id} />
        </Grid>
      ))}
    </Grid>
  );
};
