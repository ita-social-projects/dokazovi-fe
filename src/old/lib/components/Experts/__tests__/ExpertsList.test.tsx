import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ExpertsList } from '../ExpertsList';
import { IExpert } from '../../../types';

const MOCK_EMPTY_EXPERTS_LIST: IExpert[] = [];

const MOCK_EXPERTS_LIST: IExpert[] = [
  {
    id: 10,
    firstName: 'Марія',
    lastName: 'Марієнко',
    email: 'masha@mail.com',
    qualification: 'Лікар вищої категорії',
    phone: '+380956456969',
    avatar: 'https://i.pravatar.cc/300?img=16',
    bio: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 6,
        name: 'cardiology',
        label: 'Кардіологія',
        color: '#00ffff',
        hasDoctors: true,
        hasPosts: true,
      },
      {
        id: 4,
        name: 'therapy',
        label: 'Терапія',
        color: '#ffee58',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 5,
      city: {
        id: 55,
        name: 'Бровари',
      },
      name: 'Medical Idea',
    },
    lastAddedPost: {
      id: 266,
      title: 'Офтальмологія',
    },
  },
  {
    id: 6,
    firstName: 'Палана',
    lastName: 'Литвинова',
    email: 'PalanaLitvinova514@mail.com',
    qualification: 'Кандидат медичних наук',
    phone: '+380633484351',
    avatar: 'https://i.pravatar.cc/300?img=44',
    bio:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 6,
        name: 'cardiology',
        label: 'Кардіологія',
        color: '#00ffff',
        hasDoctors: true,
        hasPosts: true,
      },
      {
        id: 4,
        name: 'therapy',
        label: 'Терапія',
        color: '#ffee58',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 4,
      city: {
        id: 119,
        name: 'Дніпро',
      },
      name: 'Медікум',
    },
    lastAddedPost: {
      id: 246,
      title: 'Sit amet consectetur',
    },
  },
  {
    id: 1,
    firstName: 'Таржеман',
    lastName: 'Соколов',
    email: 't.sokolov@mail.com',
    qualification: 'Кандидат медичних наук',
    phone: '+380633335533',
    avatar: 'https://i.pravatar.cc/300?img=44',
    bio:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor ',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 4,
        name: 'therapy',
        label: 'Терапія',
        color: '#ffee58',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 4,
      city: {
        id: 119,
        name: 'Дніпро',
      },
      name: 'Медікум',
    },
    lastAddedPost: {
      id: 246,
      title: 'Sit amet consectetur',
    },
  },
  {
    id: 16,
    firstName: 'Олег',
    lastName: 'Петренко',
    email: 'petrenko@mail.com',
    qualification: 'Лікар вищої категорії',
    phone: '+380957773377',
    avatar: 'https://i.pravatar.cc/300?img=16',
    bio: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 6,
        name: 'cardiology',
        label: 'Кардіологія',
        color: '#00ffff',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 5,
      city: {
        id: 55,
        name: 'Бровари',
      },
      name: 'Medical Idea',
    },
    lastAddedPost: {
      id: 266,
      title: 'Офтальмологія',
    },
  },
  {
    id: 2,
    firstName: 'Милетий',
    lastName: 'Чернышов',
    email: 'MiletiyChernyshov229@mail.com',
    qualification: 'Лікар вищої категорії',
    phone: '+380956687969',
    avatar: 'https://i.pravatar.cc/300?img=16',
    bio: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 6,
        name: 'cardiology',
        label: 'Кардіологія',
        color: '#00ffff',
        hasDoctors: true,
        hasPosts: true,
      },
      {
        id: 4,
        name: 'therapy',
        label: 'Терапія',
        color: '#ffee58',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 5,
      city: {
        id: 55,
        name: 'Бровари',
      },
      name: 'Medical Idea',
    },
    lastAddedPost: {
      id: 266,
      title: 'Офтальмологія',
    },
  },
  {
    id: 5,
    firstName: 'Гигран',
    lastName: 'Алексеев',
    email: 'GigranAlekseev32@mail.com',
    qualification: 'Кандидат медичних наук',
    phone: '+380633484351',
    avatar: 'https://i.pravatar.cc/300?img=44',
    bio:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 4,
        name: 'therapy',
        label: 'Терапія',
        color: '#ffee58',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 4,
      city: {
        id: 119,
        name: 'Дніпро',
      },
      name: 'Медікум',
    },
    lastAddedPost: {
      id: 246,
      title: 'Sit amet consectetur',
    },
  },
  {
    id: 13,
    firstName: 'Марина',
    lastName: 'Вовк',
    email: 'maryna@mail.com',
    qualification: 'Кандидат медичних наук',
    phone: '+380633335533',
    avatar: 'https://i.pravatar.cc/300?img=44',
    bio:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor ',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 4,
        name: 'therapy',
        label: 'Терапія',
        color: '#ffee58',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 4,
      city: {
        id: 119,
        name: 'Дніпро',
      },
      name: 'Медікум',
    },
    lastAddedPost: {
      id: 246,
      title: 'Sit amet consectetur',
    },
  },
  {
    id: 7,
    firstName: 'Регіна',
    lastName: 'Регіненко',
    email: 'regina@mail.com',
    qualification: 'Лікар вищої категорії',
    phone: '+380957773377',
    avatar: 'https://i.pravatar.cc/300?img=16',
    bio: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 6,
        name: 'cardiology',
        label: 'Кардіологія',
        color: '#00ffff',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 5,
      city: {
        id: 55,
        name: 'Бровари',
      },
      name: 'Medical Idea',
    },
    lastAddedPost: {
      id: 266,
      title: 'Офтальмологія',
    },
  },
  {
    id: 12,
    firstName: 'Степан',
    lastName: 'Степанов',
    email: 'stepan@mail.com',
    qualification: 'Лікар вищої категорії',
    phone: '+380956456969',
    avatar: 'https://i.pravatar.cc/300?img=16',
    bio: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 6,
        name: 'cardiology',
        label: 'Кардіологія',
        color: '#00ffff',
        hasDoctors: true,
        hasPosts: true,
      },
      {
        id: 4,
        name: 'therapy',
        label: 'Терапія',
        color: '#ffee58',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 5,
      city: {
        id: 55,
        name: 'Бровари',
      },
      name: 'Medical Idea',
    },
    lastAddedPost: {
      id: 266,
      title: 'Офтальмологія',
    },
  },
  {
    id: 3,
    firstName: 'Незда',
    lastName: 'Ушаков',
    email: 'NezdaUshakov899@mail.com',
    qualification: 'Кандидат медичних наук',
    phone: '+380633484351',
    avatar: 'https://i.pravatar.cc/300?img=44',
    bio:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 6,
        name: 'cardiology',
        label: 'Кардіологія',
        color: '#00ffff',
        hasDoctors: true,
        hasPosts: true,
      },
      {
        id: 4,
        name: 'therapy',
        label: 'Терапія',
        color: '#ffee58',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 4,
      city: {
        id: 119,
        name: 'Дніпро',
      },
      name: 'Медікум',
    },
    lastAddedPost: {
      id: 246,
      title: 'Sit amet consectetur',
    },
  },
  {
    id: 4,
    firstName: 'Кукури',
    lastName: 'Чапко',
    email: 'KukuriChapko624@mail.com',
    qualification: 'Кандидат медичних наук',
    phone: '+380633335533',
    avatar: 'https://i.pravatar.cc/300?img=44',
    bio:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor ',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 4,
        name: 'therapy',
        label: 'Терапія',
        color: '#ffee58',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 4,
      city: {
        id: 119,
        name: 'Дніпро',
      },
      name: 'Медікум',
    },
    lastAddedPost: {
      id: 246,
      title: 'Sit amet consectetur',
    },
  },
  {
    id: 14,
    firstName: 'Олег',
    lastName: 'Петренко',
    email: 'petrenko@mail.com',
    qualification: 'Лікар вищої категорії',
    phone: '+380957773377',
    avatar: 'https://i.pravatar.cc/300?img=16',
    bio: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 6,
        name: 'cardiology',
        label: 'Кардіологія',
        color: '#00ffff',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 5,
      city: {
        id: 55,
        name: 'Бровари',
      },
      name: 'Medical Idea',
    },
    lastAddedPost: {
      id: 266,
      title: 'Офтальмологія',
    },
  },
  {
    id: 8,
    firstName: 'Тарас',
    lastName: 'Шевченко',
    email: 'taras@mail.com',
    qualification: 'Лікар-спеціаліст',
    phone: '+380633335533',
    avatar: 'https://i.pravatar.cc/300?img=44',
    bio:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor ',
    socialNetwork: 'https://www.youtube.com',
    directions: [
      {
        id: 4,
        name: 'therapy',
        label: 'Терапія',
        color: '#ffee58',
        hasDoctors: true,
        hasPosts: true,
      },
    ],
    mainInstitution: {
      id: 4,
      city: {
        id: 119,
        name: 'Вінниця',
      },
      name: 'Інномед',
    },
    lastAddedPost: {
      id: 246,
      title: 'Sit amet consectetur',
    },
  },
];

describe('ExpertsList test', () => {
  it('should render component and match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ExpertsList experts={MOCK_EXPERTS_LIST} />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('ExpertsList without data test', () => {
  it('should render error screen and match snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ExpertsList experts={MOCK_EMPTY_EXPERTS_LIST} />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render error message when no experts to show', () => {
    render(
      <MemoryRouter>
        <ExpertsList experts={MOCK_EMPTY_EXPERTS_LIST} />
      </MemoryRouter>,
    );
    expect(
      screen.getByText(
        'На жаль, даних, що відповідають вашому запиту, не знайдено.',
      ),
    ).toBeInTheDocument();
  });
});
