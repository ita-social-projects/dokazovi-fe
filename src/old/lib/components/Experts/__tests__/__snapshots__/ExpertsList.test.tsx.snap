// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpertsList test should render component and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/10"
        >
          <div
            class="MuiBox-root MuiBox-root-3 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=16"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Марія Марієнко
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Лікар вищої категорії, Medical Idea, Бровари
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/6"
        >
          <div
            class="MuiBox-root MuiBox-root-8 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=44"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Палана Литвинова
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Кандидат медичних наук, Медікум, Дніпро
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/1"
        >
          <div
            class="MuiBox-root MuiBox-root-9 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=44"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Таржеман Соколов
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Кандидат медичних наук, Медікум, Дніпро
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/16"
        >
          <div
            class="MuiBox-root MuiBox-root-10 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=16"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Олег Петренко
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Лікар вищої категорії, Medical Idea, Бровари
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/2"
        >
          <div
            class="MuiBox-root MuiBox-root-11 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=16"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Милетий Чернышов
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Лікар вищої категорії, Medical Idea, Бровари
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/5"
        >
          <div
            class="MuiBox-root MuiBox-root-12 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=44"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Гигран Алексеев
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Кандидат медичних наук, Медікум, Дніпро
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/13"
        >
          <div
            class="MuiBox-root MuiBox-root-13 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=44"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Марина Вовк
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Кандидат медичних наук, Медікум, Дніпро
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/7"
        >
          <div
            class="MuiBox-root MuiBox-root-14 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=16"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Регіна Регіненко
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Лікар вищої категорії, Medical Idea, Бровари
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/12"
        >
          <div
            class="MuiBox-root MuiBox-root-15 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=16"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Степан Степанов
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Лікар вищої категорії, Medical Idea, Бровари
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/3"
        >
          <div
            class="MuiBox-root MuiBox-root-16 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=44"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Незда Ушаков
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Кандидат медичних наук, Медікум, Дніпро
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/4"
        >
          <div
            class="MuiBox-root MuiBox-root-17 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=44"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Кукури Чапко
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Кандидат медичних наук, Медікум, Дніпро
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/14"
        >
          <div
            class="MuiBox-root MuiBox-root-18 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=16"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Олег Петренко
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Лікар вищої категорії, Medical Idea, Бровари
            </h6>
          </div>
        </a>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
    >
      <div
        class="MuiPaper-root MuiCard-root ExpertPhotoDataCard-root-1 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <a
          href="/experts/8"
        >
          <div
            class="MuiBox-root MuiBox-root-19 ExpertPhotoDataCard-photo-2"
          >
            <img
              alt="doctor"
              aria-hidden="true"
              class="makeStyles-photo-4"
              src="https://i.pravatar.cc/300?img=44"
            />
          </div>
          <div
            class="MuiPaper-root MuiCard-root ExpertDataCard-root-5 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <h4
              class="MuiTypography-root ExpertDataCard-name-6 MuiTypography-h4"
            >
              Тарас Шевченко
            </h4>
            <h6
              class="MuiTypography-root ExpertDataCard-qualification-7 MuiTypography-h6 MuiTypography-gutterBottom"
            >
              Лікар-спеціаліст, Інномед, Вінниця
            </h6>
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ExpertsList without data test should render error screen and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
    style="font-size: 24px; user-select: none; width: 400px; height: 240px; margin: 100px auto 0px;"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      style="height: 65px; width: 65px; margin-bottom: 30px;"
      viewBox="0 0 24 24"
    >
      <circle
        cx="15.5"
        cy="9.5"
        r="1.5"
      />
      <circle
        cx="8.5"
        cy="9.5"
        r="1.5"
      />
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-6c-2.33 0-4.32 1.45-5.12 3.5h1.67c.69-1.19 1.97-2 3.45-2s2.75.81 3.45 2h1.67c-.8-2.05-2.79-3.5-5.12-3.5z"
      />
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      style="font-weight: 500; font-family: Raleway; font-size: 24px; line-height: 32px;"
    >
      На жаль, даних, що відповідають вашому запиту, не знайдено.
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-btnCancel-21 makeStyles-btn-20 MuiButton-textPrimary"
      data-testid="button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        ПОВЕРНУТИСЯ НА ГОЛОВНУ
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
